lib/SPIRV/libSPIRV/SPIRVValue.cpp

```diff
@@ -64,7 +64,16 @@ void SPIRVValue::setAlignment(SPIRVWord A) {
 }
 
 bool SPIRVValue::hasAlignment(SPIRVWord *Result) const {
-  return hasDecorate(DecorationAlignment, 0, Result);
+  if (bool HasDecorate = hasDecorate(DecorationAlignment, 0, Result))
+    return HasDecorate;
+
+  SPIRVId AlignId;
+  bool HasDecorateId = hasDecorateId(DecorationAlignmentId, 0, &AlignId);
+  if (HasDecorateId) {
+    SPIRVConstant *Cst = Module->get<SPIRVConstant>(AlignId);
```

Comment:
This doesn't work with specialization constants, I'll upload a new version of this patch.